### DISCUSSION TITLE
How to use Env variables in a js file

### DISCUSSION THREAD

**Discussion by YusufcanY** [nuxt/nuxt#17718] [Questions]
How can i use my env variables in js file

```
assets/js/test.js

const rpcState = process.env.RPC_STATE
console.log(rpcState) // undefined
```

**Comment by miker059** [comment]
If you building with Vite use **import.meta.env.VITE_ RPC_STATE**. 
The prefix **VITE_** required. Read more here https://vitejs.dev/guide/env-and-mode.html

**Reply by YusufcanY** [reply]
i changed the prefix to ' ' with vite-plugin-environment, and in prod it doesnt work, all variables are undef

**Reply by YusufcanY** [reply]
@miker059 thanks
